isisd/isis_te.c

```diff
@@ -107,6 +107,8 @@ void isis_mpls_te_create(struct isis_area *area)
 		area->mta = new;
 	} else {
 		area->mta->status = enable;
+		if (!area->mta->ted)
+			area->mta->ted = ls_ted_new(1, "ISIS", 0);
```

Comment:
I think it is not the good way to proceed.

In fact, `ls_ted_del_all()` was call in `isis_mpls_te_disable()' while it should call `ls_ted_clean()`. This will avoid to delete the TED when mpls_te is disabled and then re-enabled.

Reply:
done